docs/designers-developers/developers/themes/theme-json.md

```diff
@@ -190,6 +191,42 @@ The output to be enqueued will be:
 
 The goal is that presets can be defined using this format, although, right now, the name property (used in the editor) can't be translated from this file. For that reason, and to maintain backward compatibility, the presets declared via `add_theme_support` will also generate the CSS Custom Properties. If the `experimental-theme.json` contains any presets, these will take precedence over the ones declared via `add_theme_support`.
 
+### Theme-only CSS Custom Properties
+
+Besides the presets becoming CSS Custom Properties, the theme.json also allows for themes to create their own, so they don't have to be enqueued separately. Any values declared within the `settings.custom` section will be transformed to CSS Custom Properties following this naming schema: `--wp--theme--<variable-name>`.
+
+For example, for this input:
+
+```json
+{
+  "global": {
+    "settings": {
+      "custom": {
+        "base-font": 16,
+        "line-height": {
+          "small": 1.2,
+          "medium": 1.4,
+          "large": 1.8
+        }
+      }
+    }
+  }
+}
+```
+
+The output will be:
+
+```css
+:root {
+  --wp--theme--base-font: 16;
```

Comment:
I think we don't need the theme prefix. The variables may not be added by the theme but for example by a plugin that filters theme.json. Also, core in the future may have default variables and it may be nece to merge core and theme variables.

Reply:
I'm not sure I see the use cases of plugin and core. But, in the interests of having something more general, what do you think of `--wp--custom--...` instead?

Reply:
Changed to `--wp--custom--` in https://github.com/WordPress/gutenberg/pull/25446/commits/8eba9be11c7aed7c81923ad3f1bec75abc2e360b

Reply:
Our documentation still refers --wp--theme-- should we update it?

Reply:
Absolutely! Addressed at https://github.com/WordPress/gutenberg/pull/25446/commits/a3c6b25830549551ed153590b0d0a022bb21bc9a